### ISSUE TITLE
Comparison of constraints and expressions is broken

### ISSUE THREAD

**Issue by olafmaurer** [vagmcs/Optimus#26]
It depends on the broken equals method of the trove4j hashmap to compare the terms.

Here is a minimal example:

```
implicit val mpModel: MPModel = MPModel()
val x = MPIntVar("x", 0 to 1)
val y = MPIntVar("y", 0 to 2)

2000 * x == 3000 * y 
res: Boolean = true

```

**Comment by vagmcs** [comment]
Hello @olafmaurer,

Indeed, `trove4j 3.0.3` version has some bug in the equal method for hash maps. Luckily the issue has been fixed in the newer version `3.1.0`. So I updated the trove dependency and now it seems OK. Checkout commit 20079b0515ccfe4bffc99dbc78c7fd70d0930a58.

**Note** that the issue has been fixed in the `develop` branch. I will try and release the next version of Optimus in the following days.
